build.gradle

```diff
@@ -1,101 +1,162 @@
+
+    roborioDebug wpi.java.deps.wpilibJniDebug(wpi.platforms.roborio)
+    roborioDebug wpi.java.vendor.jniDebug(wpi.platforms.roborio)
+
+    roborioRelease wpi.java.deps.wpilibJniRelease(wpi.platforms.roborio)
+    roborioRelease wpi.java.vendor.jniRelease(wpi.platforms.roborio)
+
+    nativeDebug wpi.java.deps.wpilibJniDebug(wpi.platforms.desktop)
+    nativeDebug wpi.java.vendor.jniDebug(wpi.platforms.desktop)
+    simulationDebug wpi.sim.enableDebug()
+
+    nativeRelease wpi.java.deps.wpilibJniRelease(wpi.platforms.desktop)
+    nativeRelease wpi.java.vendor.jniRelease(wpi.platforms.desktop)
+    simulationRelease wpi.sim.enableRelease()
+
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.10.1'
+    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
+}
+
+test {
+    useJUnitPlatform()
+    systemProperty 'junit.jupiter.extensions.autodetection.enabled', 'true'
+}
+
+// Simulation configuration (e.g. environment variables).
+wpi.sim.addGui().defaultEnabled = true
+wpi.sim.addDriverstation()
+
+// Setting up my Jar File. In this case, adding all libraries into the main jar ('fat jar')
+// in order to make them all available at runtime. Also adding the manifest so WPILib
+// knows where to look for our Robot Class.
+jar {
+    from { configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } }
+    from sourceSets.main.allSource
+    manifest edu.wpi.first.gradlerio.GradleRIOPlugin.javaManifest(ROBOT_MAIN_CLASS)
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
+}
+
+// Configure jar and deploy tasks
+deployArtifact.jarTask = jar
+wpi.java.configureExecutableTasks(jar)
+wpi.java.configureTestTasks(test)
+
+// Configure string concat to always inline compile
+tasks.withType(JavaCompile) {
+    options.compilerArgs.add '-XDstringConcat=inline'
+}
+
+def getRobotVersionData() { 
+    def robotVersion = new StringBuilder()     
+    robotVersion.append("Commit: @"+"git rev-parse --short HEAD".execute().text.replace('\n', '')+" by "+("git config --list".execute().text).split("user.name=")[1].split("\n")[0]+"\n")       
+    robotVersion.append("Branch: ")
+    robotVersion.append('git branch --show-current'.execute().text)
+    robotVersion.append("Uncommitted Changes: ")
+    robotVersion.append('git diff --shortstat'.execute().text)
+    robotVersion.append("--------------------\n")
+    robotVersion.append('git status -s'.execute().text)
+    return robotVersion.toString()
+}
+
+def getMinifiedRobotVersionData() {
+    def miniData = new StringBuilder()
+    miniData.append(("git config --list".execute().text).split("user.name=")[1].split("\n")[0]+"@")
+    miniData.append("git branch --show-current".execute().text)
+    miniData.append(" | "+"git rev-parse --short HEAD".execute().text)
+    miniData.append(" | "+("git status".execute().text.length() > 0 ? "*" : ""))
+    String miniDataString = miniData.toString().replace('\n', '') 
+    return miniDataString 

```

Comment:
May as well just directly return rather than making a variable that just gets returned on the next line.

Reply:
👍